### ISSUE TITLE
Recursive Build Error

### ISSUE THREAD

**Issue by Tech1k** [LucentCoin/Lucent#13]
OS: Ubuntu 18.04 LTS
All dependencies installed
Commands ran (in order):
```
./autogen.sh
./configure --with-gui=no --disable-tests
sudo make install
```

## Log:
```
Tech1k@###-##-##-###:~/lucent$ sudo make install
Making install in src
make[1]: Entering directory '/home/ubuntu/lucent/src'
make[2]: Entering directory '/home/ubuntu/lucent/src'
  CXX      crypto/liblucentconsensus_la-aes.lo
  CXX      crypto/liblucentconsensus_la-hmac_sha256.lo
  CXX      crypto/liblucentconsensus_la-hmac_sha512.lo
  CXX      crypto/liblucentconsensus_la-ripemd160.lo
  CC       crypto/liblucentconsensus_la-aes_helper.lo
  CXX      crypto/liblucentconsensus_la-sha1.lo
  CXX      crypto/liblucentconsensus_la-sha256.lo
  CXX      crypto/liblucentconsensus_la-sha512.lo
  CC       crypto/liblucentconsensus_la-blake.lo
  CC       crypto/liblucentconsensus_la-bmw.lo
  CC       crypto/liblucentconsensus_la-cubehash.lo
  CC       crypto/liblucentconsensus_la-echo.lo
  CC       crypto/liblucentconsensus_la-groestl.lo
  CC       crypto/liblucentconsensus_la-jh.lo
  CC       crypto/liblucentconsensus_la-keccak.lo
  CC       crypto/liblucentconsensus_la-luffa.lo
  CC       crypto/liblucentconsensus_la-shavite.lo
  CC       crypto/liblucentconsensus_la-simd.lo
  CC       crypto/liblucentconsensus_la-skein.lo
  CXX      liblucentconsensus_la-arith_uint256.lo
  CXX      consensus/liblucentconsensus_la-merkle.lo
  CXX      liblucentconsensus_la-hash.lo
  CXX      primitives/liblucentconsensus_la-block.lo
  CXX      primitives/liblucentconsensus_la-transaction.lo
  CXX      liblucentconsensus_la-pubkey.lo
  CXX      script/liblucentconsensus_la-lucentconsensus.lo
  CXX      script/liblucentconsensus_la-interpreter.lo
  CXX      script/liblucentconsensus_la-script.lo
  CXX      script/liblucentconsensus_la-script_error.lo
  CXX      liblucentconsensus_la-uint256.lo
  CXX      liblucentconsensus_la-utilstrencodings.lo
make[3]: Entering directory '/home/ubuntu/lucent/src/secp256k1'
gcc -I. -g -O2 -Wall -Wextra -Wno-unused-function -c src/gen_context.c -o gen_context.o
gcc gen_context.o -o gen_context
./gen_context
  CC       src/libsecp256k1_la-secp256k1.lo
  CCLD     libsecp256k1.la
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
make[3]: Leaving directory '/home/ubuntu/lucent/src/secp256k1'
  CXXLD    liblucentconsensus.la
/usr/bin/ar: `u' modifier ignored since `D' is the default (see `U')
  CXX      lucentd-lucentd.o
  CXX      liblucent_server_a-activemasternode.o
  CXX      liblucent_server_a-addrman.o
  CXX      liblucent_server_a-addrdb.o
  CXX      liblucent_server_a-alert.o
  CXX      liblucent_server_a-bloom.o
  CXX      liblucent_server_a-blockencodings.o
  CXX      liblucent_server_a-chain.o
  CXX      liblucent_server_a-checkpoints.o
  CXX      liblucent_server_a-dsnotificationinterface.o
  CXX      liblucent_server_a-httprpc.o
  CXX      liblucent_server_a-httpserver.o
  CXX      liblucent_server_a-init.o
g++: internal compiler error: Killed (program cc1plus)
Please submit a full bug report,
with preprocessed source if appropriate.
See <file:///usr/share/doc/gcc-7/README.Bugs> for instructions.
Makefile:6440: recipe for target 'liblucent_server_a-init.o' failed
make[2]: *** [liblucent_server_a-init.o] Error 4
make[2]: Leaving directory '/home/ubuntu/lucent/src'
Makefile:10568: recipe for target 'install-recursive' failed
make[1]: *** [install-recursive] Error 1
make[1]: Leaving directory '/home/ubuntu/lucent/src'
Makefile:688: recipe for target 'install-recursive' failed
make: *** [install-recursive] Error 1
```

**Comment by AlphaSerpentis** [comment]
Can you try cleaning up the project and starting from `cd depends`, `sudo make`, and then `cd .. && ./autogen.sh && ./configure <ARGUMENTS>` and then finally `sudo make && sudo make install`

**Comment by AlphaSerpentis** [comment]
Confirmation?

**Comment by Tech1k** [comment]
Sorry, been busy. I'll try that today.

**Comment by Tech1k** [comment]
Just tested with master, the issue has been resolved with #15.
